### PR TITLE
[feat] Break down attention mechanism. Add figures

### PR DESCRIPTION
**Pull request by tmynn**
Added 2 figures to illustrate K, Q, and V matrices formation, and showcase the attention operation. Gave explanation of the key, query, and value terms with the analogy of retrieval mechanism.

### CODE REVIEW

**Comment by israfelsr** [comment]
I would modify some of the names of the dimensions to make it more clear. Something like:
B: batch size (I wouldn't add this)
C: context sequence lenght
E: embedding dimension
H: hidden dimension
Q: query sequence lenght
I would also make the drawings about cross-attention using a different Q and C and then just mention that self-attention uses the same input for both, query and context. 

Also there are two small errors in the equation, the operation inside the softmax is $Q \cdot K^T$. The dimension there should be BxQxC. 
Then is $sofmax(\cdot)\cdot V$, and you end up with BxQxH.

File: pages/transformers.tex [hunk 1]
```diff
@@ -594,15 +627,15 @@ \subsection{Training with Encoder-Decoder}
 # Random prompt, uses pooling
 for i in range(5):
     set_seed(42)
-    model.prompt("Barack Obama, the", 50, 3)
+    model.prompt("Ramon Astudillo, the", 50, 3)
```
Comment (ramon-astudillo) [inline pages/transformers.tex:630]: can you add "Barack Obama" instead? :)

**Review by ramon-astudillo — APPROVED** [review]
